wallet/src/Pos/Wallet/Web/Methods/Misc.hs

```diff
@@ -91,6 +93,21 @@ syncProgress =
     <*> connectedPeers
 
 ----------------------------------------------------------------------------
+-- NTP (Network Time Protocol) based time difference
+----------------------------------------------------------------------------
+
+localTimeDifference :: MonadWalletWebMode m => m Word
+localTimeDifference =
+    mkNtpStatusVar >>= readMVar >>= pure . diff
+  where
+    diff :: NtpStatus -> Word
+    diff = \case
+        NtpSyncOk -> 0
```

Comment:
Well this doesn't seem fully accurate.

My understanding if it is `NtpSyncOk` that doesn't mean `localTimeDifference == 0` but it means `localTimeDifference < timeDifferenceWarnThreshold` which isn't the same.

Reply:
Also, I am not sure we would like to use `timeDifferenceWarnThreshold` for this purpose (which is to notify user via UI) , or we should define another constant `timeDifferenceUserWarningThreshold` and warn the user to adjust its clock before (ie, user threshold should be smaller then `timeDifferenceWarnThreshold`) it actually hits the limit.

@pva701 @flyingleafe thoughts ?

Reply:
> if it is NtpSyncOk that doesn't mean localTimeDifference == 0

@akegalj ^ That's correct. However, the first idea of this endpoint is to keep the returned value as simple as possible. Which means: `Daedalus` is interested in a value of time difference if we have `NtpDesync` only. If we have `NtpSyncOk` we won't return a time difference, even if the difference (`d`) is `d >= 0 && d < timeDifferenceWarnThreshold`. In this case we just send a value of `0` to hide the computation by using `timeDifferenceWarnThreshold` and just to say that everything is ok.

If there is a need, we can change the returned value of this endpoint of course ;)

Reply:
sounds reasonable to me